Support for specialized functions

haskellcamargo opened the pull request:
This pull-request implements support for specialized calls via `R`. Soon, a good improvement will be making the a parameter, but it solves the problem for now. I suggest using `2.1.0` as version name for this.

Now, both `R.map(R.prop)` and `map(prop)` can be recognized and works mutually. Adapting to check the scope later also became easier; I've added an `ast-helper` to get the work done. We currently have `135` tests.

This introduces no breaking changes.
This introduces no new rules.

Closes #7.
Thanks @polytypic for reporting and @lo1tuma for the idea.

lo1tuma commented on ast-helper.js:
```diff
@@ -0,0 +1,35 @@
+);
+
+// :: String -> Node -> Boolean
+const isName = name => R.either(
```
I think `isName` is not very expressive, I would suggest something like `isRamdaMethod`.

lo1tuma commented on rules/any-pass-simplification.js:
```diff
@@ -1,24 +1,24 @@
+            arguments: R.both(
+                R.propSatisfies(R.lt(0), 'length'),
+                R.where({
+                    0: R.propSatisfies(R.all(isCalling({ name: 'complement' })), 'elements')
```
What happens when the first argument is not an `ArrayExpression`, e.g. `anyPass(myPredicates, myData)`? Looks like the rule would crash in this case.

lo1tuma commented on rules/any-pass-simplification.js:
```diff
@@ -1,24 +1,24 @@
 'use strict';
+const R = require('ramda');
+const isCalling = require('../ast-helper').isCalling;
```
This file needs to be added to the whitelist of published files in `package.json`.

haskellcamargo replied:
Fixed.

lo1tuma commented on rules/cond-simplification.js:
```diff
@@ -1,12 +1,20 @@
+            arguments: R.both(
+                R.propSatisfies(R.lt(0), 'length'),
+                R.where({
+                    0: R.pathSatisfies(R.gte(2), ['elements', 'length'])
```
Same here, the first argument might not be an `ArrayExpression`.

haskellcamargo commented:
Fixed everything listed!

I couldn't think of a better name than `isCalling`, so I kept it for now.
I've implemented support for literal array access, such as `R['filter']` too.
The line and columns are reported by `node`, which is passed via the parent call, so, for now, it will always link to the call expression correctly, but we can further add test cases for that.
